packages/block-library/src/navigation-link/edit.js

```diff
@@ -108,6 +108,7 @@ function NavigationLinkEdit( {
 							setAttributes( { description: descriptionValue } );
 						} }
 						label={ __( 'Description' ) }
+						help={ __( 'The description will be displayed in the menu if the current theme supports it.' ) }
```

Comment:
Where does the description end up? In a title attribute?

Reply:
On the Maywood theme (and I'm guessing most others), it doesn't show up on the frontend at all.

Reply:
It's not being used at all: https://github.com/WordPress/gutenberg/blob/master/packages/block-library/src/navigation/index.php#L126:L165
We've given users a dead end.

Reply:
It looks like this was added for parity with Core's menu-item post type: https://core.trac.wordpress.org/browser/tags/5.3.2/src/wp-content/themes/twentyfifteen/functions.php#L500